platforms/software/dependency-management/src/main/java/org/gradle/internal/component/local/model/DefaultProjectComponentSelector.java

```diff
@@ -60,7 +61,7 @@ public DefaultProjectComponentSelector(BuildIdentifier buildIdentifier, Path ide
     @Override
     public String getDisplayName() {
         String prefix;
-        if (identityPath == Path.ROOT) {
+        if (Objects.equals(identityPath, Path.ROOT)) {
```

Comment:
Why use Objects.equals?

Reply:
Vs. `identityPath.equals`, you mean?  I just always use that in case of nulls ever appearing on the LHS.

Reply:
identityPath will never be null for a project dependency. Plus even if it could be Path.ROOT is never null

Reply:
True, I'm just being paranoid and future-proofing.  This isn't performance critical code.

Reply:
Yeah I'm not worried about performance at all, mostly a readability thing. If I see code like this I immediately thing that one of the parameters _could_ be nullable, so to me its confusing to see it.